### PR TITLE
Pi 1125 fix authentication get token method nk

### PR DESCRIPTION
**Pull request by natke**
Fix method for authentication getToken (get --> put) (PI-1125)
Rename readerId to readerName as per the ItemSense server (PI-1184)
Remove unused logger in RestApiHelper
Remove unused RestApiHelper for gets in JobResponse
Add more tests for JobController.getJobs
Add variable for jackson version in POM to make upgrading this library easier
Updated major version to indicate incompatibility with previous version (PI-1118)
Updated README

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -70,12 +71,6 @@
         </dependency>
 
         <dependency>
-            <groupId>com.google.code.gson</groupId>
```
Comment (davidyscott) [inline pom.xml]: Don't the tests use gson?

File: src/main/java/com/impinj/itemsense/client/coordinator/authentication/AuthenticationController.java [hunk 2]
```diff
@@ -23,7 +23,7 @@ public Response getTokenAsResponse() {
     }
 
     public Response getTokenAsResponse(String username) {
-        return this.restApiHelper.get(target, "/authentication/v1/token", username);
```
Comment (davidyscott) [inline src/main/java/com/impinj/itemsense/client/coordinator/authentication/AuthenticationController.java]: ...awkward

File: src/main/java/com/impinj/itemsense/client/coordinator/job/JobController.java [hunk 3]
```diff
@@ -18,29 +18,27 @@
...
 
     private WebTarget target;
-    private RestApiHelper<Job> restApiHelperPoster;
-    private RestApiHelper<JobResponse> restApiHelperGetter;
+    private RestApiHelper<Job> restApiHelper;
 
     public JobController(final WebTarget target) {
         this.target = target;
-        this.restApiHelperPoster = new RestApiHelper<>(Job.class);
-        this.restApiHelperGetter = new RestApiHelper<>(JobResponse.class);
```
Comment (davidyscott) [inline src/main/java/com/impinj/itemsense/client/coordinator/job/JobController.java]: Does this work???

File: src/main/java/com/impinj/itemsense/client/coordinator/job/JobController.java [hunk 4]
```diff
@@ -18,29 +18,27 @@
...
+        this.restApiHelper = new RestApiHelper<>(Job.class);
```
Comment (davidyscott) [inline src/main/java/com/impinj/itemsense/client/coordinator/job/JobController.java]: Does this work???

Comment (natke) [inline src/main/java/com/impinj/itemsense/client/coordinator/job/JobController.java]: Yes. The reader does not use the templated class.

File: pom.xml (hunk 1 above)
Comment (natke) [inline pom.xml]: I'll check. It does run and build, but perhaps the other dependencies pull it in.

File: src/main/java/com/impinj/itemsense/client/coordinator/authentication/AuthenticationController.java (hunk 2 above)
Comment (natke) [inline src/main/java/com/impinj/itemsense/client/coordinator/authentication/AuthenticationController.java]: Right

File: src/main/java/com/impinj/itemsense/client/coordinator/job/JobController.java (hunk 4 above)
Comment (davidyscott) [inline src/main/java/com/impinj/itemsense/client/coordinator/job/JobController.java]: Ah, that's right.  Sorry!
Hm, looking at this again...  it doesn't look like the type parameter serves any purpose?

Comment (natke) [inline src/main/java/com/impinj/itemsense/client/coordinator/job/JobController.java]: I think you're right. I will test it.

File: pom.xml (hunk 1 above)
Comment (natke) [inline pom.xml]: The answer is yes, they do use it, but the net.dongliu:gson-java8-datatype is pulling it in. I will add it back in with a scope of test

File: src/main/java/com/impinj/itemsense/client/coordinator/job/JobController.java (hunk 4 above)
Comment (natke) [inline src/main/java/com/impinj/itemsense/client/coordinator/job/JobController.java]: You are right. It is not used. I've removed it.

File: pom.xml (hunk 1 above)
Comment (davidyscott) [inline pom.xml]: Fun times, thanks for checking!

File: src/main/java/com/impinj/itemsense/client/helpers/RestApiHelper.java [hunk 5]
```diff
@@ -17,12 +13,6 @@
 import javax.ws.rs.core.Response;
 
 public class RestApiHelper<T> {
```
Comment (davidyscott) [inline src/main/java/com/impinj/itemsense/client/helpers/RestApiHelper.java]: Oh, interesting, actually my comment earlier (which github made disappear?) was about the type parameter being unnecessary, not the constructor argument, though I suppose they're pretty closely related.

Comment (natke) [inline src/main/java/com/impinj/itemsense/client/helpers/RestApiHelper.java]: The constructor argument was only used to set the parameter, hence I could remove it
The generic parameter was being inferred from the type parameter (hence having to explicitly specify these)

Comment (davidyscott) [inline src/main/java/com/impinj/itemsense/client/helpers/RestApiHelper.java]: Sure, but I don't think that parameter is being used for anything, either?

Comment (davidyscott) [inline src/main/java/com/impinj/itemsense/client/helpers/RestApiHelper.java]: But as far as I can see, it's only used as the parameter type, so all instances of `T` could just as well be replaced with `Object`?  Am I missing something?

Comment (natke) [inline src/main/java/com/impinj/itemsense/client/helpers/RestApiHelper.java]: You're right. I've removed it, and made all of the RestApiHelper methods static.

**Comment by davidyscott** [comment]
Awesome, this looks so much better!  :)

**Pull request merged**